src/components/Input/TimeInput.js

```diff
@@ -187,6 +187,10 @@ export default class TimeInput extends React.Component {
 
     const [hasTypedTens, hasTypedMin] = userInputProgress(inputCandidate, time);
 
+    if (input === '' && unavailableTimes?.has(label)) {
+      return false;
+    }
```

Comment:
What's the desired behavior here? This will hide the time **only** when the user hasn't typed anything yet. Once the user starts typing, they can still enter the time?

Reply:
Generally speaking, I would suggest checking against the `value` instead of the `label`

Reply:
> What's the desired behavior here? This will hide the time **only** when the user hasn't typed anything yet. Once the user starts typing, they can still enter the time?

yes, that's the expected behavior.

Reply:
Honestly it seems a little odd to me to support this behavior here because this use case is so specific and this is meant to be a fairly reusable component.

Personally I think one way forward for this is to split up this component into multiple functions / hooks and then expose those as part of this package and reuse them elsewhere.


If we were to reuse this component maybe a prop for determining the initially displayed times would make more sense? This isn't really "unavailable" times because the user can still select them.

Reply:
Yes I can see the prop name can be a little confusing, we could change it to be 'initiallyHiddenTimes' if that sounds more clear. we could also add notes/comments.